compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1101,7 +1101,7 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
               js.JSNew(jsSuperClassRef, args)
           }
 
-          val selfVarDef = js.VarDef(selfName, thisOriginalName, jstpe.AnyType, mutable = false, newTree)
+          val selfVarDef = js.VarDef(selfIdent, thisOriginalName, jstpe.AnyType, mutable = false, newTree)
```

Comment:
On the flipside: This will *not* change the position of the `this` ident to the position of the super call.

Reply:
Indeed. I added a `.copy()` to get the correct position here.